pkgs/flojoy/flojoy/flojoy_cloud.py

```diff
@@ -1,475 +1,344 @@
-        response = requests.request("GET", url, headers=self.headers)
-        response = json.loads(response.text)
-
-        return response
-
-    def rename_measurement(self, meas_id: str, name: str):
-        """
-        Rename the specified measurement.
-        """
-        url = f"{self.base_url}/measurements/{meas_id}"
-        payload = payload = json.dumps({"name": name})
-        response = requests.request("PATCH", url, headers=self.headers, data=payload)
-        response = json.loads(response.text)
-
-        return response
-
-    def store_dc(self, data, dc_type: str, meas_id: str, name: str):
-        """
-        A method that stores a formatted data payload in a measurement.
-        """
-        url = f"{self.base_url}/dcs/add/{meas_id}"
-        payload = self._create_payload(data, dc_type, name)
-        response = requests.request("POST", url, headers=self.headers, data=payload)
-
-        return json.loads(response.text)
+                return model.model_validate_json(json_str)
+
+    @overload
+    def __req(
+        self,
+        method: HttpMethod,
+        endpoint: str,
+        *,
+        query_params: dict[str, int | str] | None = None,
+        data: str | None = None,
+        body: dict | None = None,
+        parse_as: type[T],
+    ) -> T:
+        ...
+
+    @overload
+    def __req(
+        self,
+        method: HttpMethod,
+        endpoint: str,
+        *,
+        query_params: dict[str, int | str] | None = None,
+        data: str | None = None,
+        body: dict | None = None,
+        parse_as: TypeAdapter[U],
+    ) -> U:
+        ...
+
+    @overload
+    def __req(
+        self,
+        method: HttpMethod,
+        endpoint: str,
+        *,
+        query_params: dict[str, int | str] | None = None,
+        data: str | None = None,
+        body: dict | None = None,
+        parse_as: None = None,
+    ) -> None:
+        ...
+
+    def __req(
+        self,
+        method: HttpMethod,
+        endpoint: str,
+        *,
+        query_params: dict[str, int | str] | None = None,
+        data: str | None = None,
+        body: dict | None = None,
+        parse_as: type[T] | TypeAdapter[U] | None = None,
+    ) -> T | U | None:
+        res = requests.request(
+            method,
+            self.base_url + endpoint + self.__make_query_string(query_params),
+            data=data,
+            json=body,
+            headers={"Authorization": self.auth, "Content-Type": "application/json"},
+        )
+
+        if res.status_code >= 400:
+            error_json = json.loads(res.text)
+            raise FlojoyCloudException(
+                f"An exception occurred when making a request to {endpoint}\n "
+                f"{error_json['code']}: {error_json['message']}"
+            )
+
+        if parse_as is None:
+            return None
+
+        return self.__parse(parse_as, res.text)
+
+    """Test Endpoints"""
+
+    def create_test(
```

Comment:
https://www.python-httpx.org/advanced/#sharing-configuration-across-requests
With httpx you can share client like this